lib/IRGen/GenExistential.cpp

```diff
@@ -1016,21 +1006,31 @@ class ClassExistentialTypeInfo final
 
   TypeLayoutEntry *buildTypeLayoutEntry(IRGenModule &IGM,
                                         SILType T) const override {
-    ScalarKind kind;
-    switch (Refcounting) {
-    case ReferenceCounting::Native:
-      kind = ScalarKind::NativeStrongReference;
-      break;
-    case ReferenceCounting::ObjC:
-      kind = ScalarKind::ObjCReference;
-      break;
-    case ReferenceCounting::Unknown:
-      kind = ScalarKind::UnknownReference;
-      break;
-    default:
-      llvm_unreachable("Unsupported refcounting style");
+    // We can't create an objc typeinfo by itself, so don't destructure if we
+    // have one
+    if (Refcounting != ReferenceCounting::Native) {
+      return IGM.typeLayoutCache.getOrCreateTypeInfoBasedEntry(*this, T);
```

Comment:
As far as I can find, there is no way to synthesize a Objc Refcounted pointer TypeInfo similar to IGM.getNativeObjectTypeInfo below, so I've set it to fall back to creating a typeinfobased entry here.

Reply:
Seems reasonable.